geode-apis-compatible-with-redis/src/test/java/org/apache/geode/redis/internal/data/RedisSortedSetTest.java

```diff
@@ -0,0 +1,236 @@
+import org.apache.geode.internal.serialization.SerializationContext;
+import org.apache.geode.internal.size.ReflectionObjectSizer;
+import org.apache.geode.redis.internal.netty.Coder;
+
+public class RedisSortedSetTest {
+  private final ReflectionObjectSizer reflectionObjectSizer = ReflectionObjectSizer.getInstance();
+
+  @BeforeClass
+  public static void beforeClass() {
+    InternalDataSerializer.getDSFIDSerializer().registerDSFID(
+        DataSerializableFixedID.REDIS_SORTED_SET_ID,
+        RedisSortedSet.class);
+  }
+
+  @Test
+  public void confirmToDataIsSynchronized() throws NoSuchMethodException {
+    assertThat(Modifier.isSynchronized(RedisSortedSet.class
+        .getMethod("toData", DataOutput.class, SerializationContext.class).getModifiers()))
+            .isTrue();
+  }
+
+  @Test
+  public void confirmSerializationIsStable() throws IOException, ClassNotFoundException {
+    RedisSortedSet o1 = createRedisSortedSet("k1", "v1", "k2", "v2");
+    o1.setExpirationTimestampNoDelta(1000);
+    HeapDataOutputStream out = new HeapDataOutputStream(100);
+    DataSerializer.writeObject(o1, out);
+    ByteArrayDataInput in = new ByteArrayDataInput(out.toByteArray());
+    RedisSortedSet o2 = DataSerializer.readObject(in);
+    assertThat(o2.equals(o1)).isTrue();
+  }
+
+  @Test
+  public void equals_returnsFalse_givenDifferentExpirationTimes() {
+    RedisSortedSet o1 = createRedisSortedSet("k1", "v1", "k2", "v2");
+    o1.setExpirationTimestampNoDelta(1000);
+    RedisSortedSet o2 = createRedisSortedSet("k1", "v1", "k2", "v2");
+    o2.setExpirationTimestampNoDelta(999);
+    assertThat(o1).isNotEqualTo(o2);
+  }
+
+  @Test
+  public void equals_returnsFalse_givenDifferentValueBytes() {
+    RedisSortedSet o1 = createRedisSortedSet("k1", "v1", "k2", "v2");
+    o1.setExpirationTimestampNoDelta(1000);
+    RedisSortedSet o2 = createRedisSortedSet("k1", "v1", "k2", "v3");
+    o2.setExpirationTimestampNoDelta(1000);
+    assertThat(o1).isNotEqualTo(o2);
+  }
+
+  @Test
+  public void equals_returnsTrue_givenEqualValueBytesAndExpiration() {
+    RedisSortedSet o1 = createRedisSortedSet("k1", "v1", "k2", "v2");
+    o1.setExpirationTimestampNoDelta(1000);
+    RedisSortedSet o2 = createRedisSortedSet("k1", "v1", "k2", "v2");
+    o2.setExpirationTimestampNoDelta(1000);
+    assertThat(o1).isEqualTo(o2);
+  }
+
+  @Test
+  public void equals_returnsTrue_givenDifferentEmptySortedSets() {
+    RedisSortedSet o1 = new RedisSortedSet(Collections.emptyList());
+    RedisSortedSet o2 = NullRedisDataStructures.NULL_REDIS_SORTED_SET;
+    assertThat(o1).isEqualTo(o2);
+    assertThat(o2).isEqualTo(o1);
+  }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void zadd_stores_delta_that_is_stable() throws IOException {
+    Region<RedisKey, RedisData> region = Mockito.mock(Region.class);
+    RedisSortedSet o1 = createRedisSortedSet("k1", "v1", "k2", "v2");
+
+    ArrayList<byte[]> adds = new ArrayList<>();
```

Comment:
This should be `List<byte[]> adds = new ArrayList<>();` since it's best practice to always use interfaces rather than concrete types wherever possible.

Reply:
Good point, polished up.